pipeline_steps/common.groovy

```diff
@@ -1736,6 +1736,12 @@ void stdJob(String hook_dir, String credentials, String jira_project_key, String
               }
             } catch (Exception e){
               currentBuild.result = "FAILURE"
+              errString = "Caught exception on ${env.NODE_NAME}: ${e} Build: ${env.BUILD_URL}"
```

Comment:
Ok, so why even have the catch clause here? Will refactor based on your suggestion: implement as a try/finally only and add the nodehold logic to the outer try/catch.

Reply:
If you look at the [commit referenced in the issue](https://github.com/rcbops/rpc-gating/commit/fc1d1b33f475a7483c3a37b70391686db13f2f44), the catch block was added so that the post stage can determine whether or not the build failed during the run stage. If you remove it you need to manage that some other way.

If the exception is thrown or not caught in the first place `use_node` in `pipeline_steps/common.groovy` should be sufficient to manage the hold of the node.

Reply:
Understood. I missed the build result usage in the `finally` block.